crates/prune/src/pruner.rs

```diff
@@ -135,7 +168,7 @@ impl<DB: Database> Pruner<DB> {
         &mut self,
         provider: &DatabaseProviderRW<DB>,
         tip_block_number: BlockNumber,
-        mut delete_limit: usize,
+        mut limiter: PruneLimiter,
     ) -> Result<(PrunerStats, usize, PruneProgress), PrunerError> {
         let static_file_segments = self.static_file_segments();
         let segments = static_file_segments
```

Comment:
we should make this a ring, so the last pruned segment is first on next call. it should already have happened without the timeout. since otherwise we risk pruning the transactions but not the headers and less likely the receipts. @shekhirin , thoughts?

Reply:
agreed, that makes sense. Let's do it in a different PR

Reply:
https://github.com/paradigmxyz/reth/issues/7343